xbmc/video/windows/GUIWindowFullScreen.cpp

```diff
@@ -473,3 +474,11 @@ bool CGUIWindowFullScreen::HasVisibleControls()
 {
   return m_controlStats->nCountVisible > 0;
 }
+
+int CGUIWindowFullScreen::GetOSDWindowID()
```

Comment:
I refactored this slightly, but there's no callback for when the playing item changes from game to video, so it still calls out to `g_application` (as does every other function in this class). Does it look ok now?